ZhuoHeiChaAPI/Services/GameService.cs

```diff
@@ -11,13 +12,57 @@ namespace ZhuoHeiChaAPI.Services
 {
     public class GameService : IGameService
     {
+        private Dictionary<int, Dictionary<int, List<PayerInfo>>> _returnTable = new Dictionary<int, Dictionary<int, List<PayerInfo>>>();
+
         private int _gameCounter;
         private readonly ConcurrentDictionary<int, (IGame, object)> _gameSessions = new ConcurrentDictionary<int, (IGame, object)>();
 
         private readonly ILogger<GameService> _logger;
         private readonly ICardHelper _cardHelper;
         private readonly IGameFactory _gameFactory;
 
+
+
+        public void InitReturnTable(int gameId, Dictionary<int, IEnumerable<int>> returnTributeListByPlayerId, Dictionary<int, IEnumerable<int>> cardsToBeReturnCount) 
+        {
+            _returnTable[gameId] = new Dictionary<int, List<PayerInfo>>();
+
+            foreach (var receiverId in returnTributeListByPlayerId.Keys) 
+            {
+                
+                _returnTable[gameId][receiverId] = new List<PayerInfo>();
+
+                var ReturnTributeList = returnTributeListByPlayerId[receiverId].ToList();
+                var cardsToBeReturnCountList = cardsToBeReturnCount[receiverId].ToList();
+                
+                for (int i = 0; i < ReturnTributeList.Count; i++) 
+                {
+                    _returnTable[gameId][receiverId].Add(new PayerInfo
+                    {
+                        PayerId = ReturnTributeList[i],
+                        IsFinishedReturnTribute = false,
+                        ReturnTributeCount = cardsToBeReturnCountList[i]
+                    });
+                }
+            }
+                
+        }
+
+        public PayerInfo GetFirstPayerTarget(int gameId, int receiverId)
```

Comment:
Now that I am thinking about it, we may actually remove this method and use `GetNextPayerTarget` instead, because the first time we call this method, all `PayerInfo` object will have the property `IsFinishedReturnTribute` set to false. Therefore, calling `GetNextPayerTarget` for the very first time will indeed return the first element.

Reply:
Done